File: src/main/java/io/tbbc/cf/turret/chassis/TurretChassis.java
package io.tbbc.cf.turret.chassis;

import io.tbbc.cf.common.Size;
import io.tbbc.cf.turret.chassis.skin.ChassisSkin;

import java.util.List;

public record TurretChassis(String name,
                            String label,
                            Size size,
                            ChassisType type,
                            IChassisProps props,
                            List<ChassisSkin> availableSkins) {
}

File: src/main/webui/src/model/turret/TurretChassis.ts
import Properties from "../common/Properties";
import Size from "../common/Size";
import ChassisSkin from "./ChassisSkin";

export default class ChassisTurret {
    name: string;
    label: string;
    type: string;
    props: Properties;
    skin: string;
    availableSkins: ChassisSkin[];
    size: Size;
    constructor(name: string, label: string, type: string, props: Properties, skin: string, availableSkins: ChassisSkin[],
        size: Size) {
        this.name = name;
        this.label = label;
        this.type = type;
        this.props = props;
        this.skin = skin;
        this.availableSkins = availableSkins;
        this.size = size;
    }
}